### PR TITLE
CIF-1081 - Client side product price loading does not respect multi store hierarchy configuration

### PR DESCRIPTION
**Pull request by herzog31**
## Description

* Added `StoreViewExporter` model to expose Magento store view in DOM.
* Updated `CommerceGraphqlAPI` library to pass the `Store` header accordingly.

## How Has This Been Tested?

* Updated unit tests.

## Verification
* Open a product or category page and verify that the client-side GraphQL request that retrieves pricing information contains a `Store: default` header.
* Configure the website to use a different storeview, e.g. `venia`. You can do that by setting the `cq:magentoStore` property on a page. Verify that the client-side pricing requests now send a `Store: venia` header.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by LSantha** [comment]
@herzog31 , in case you want to avoid the overhead of the delegation pattern you can put the storeview property in a small targeted custom model that you can use in HTL to add the new property to the page source. After all you don't need to customize anything in the OOTB Page model and the delegation overhead largely surpasses the amount of relevant code. Just my 2 cents.

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/page/PageImpl.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import com.day.cq.commons.inherit.InheritanceValueMap;
+
+@Model(
+    adaptables = SlingHttpServletRequest.class,
+    adapters = { Page.class },
+    resourceType = PageImpl.RESOURCE_TYPE)
+public class PageImpl implements Page {
+
+    protected static final String RESOURCE_TYPE = "core/cif/components/structure/page/v1/page";
+    private static final String STORE_CODE_PROPERTY = "cq:magentoStore";
```
Comment (LSantha) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/page/PageImpl.java]: Should we consider renaming this JCR property to cq:magentoStoreView sometime? In Magento there a store can have multiple store views. I think we actually use this property for the store view.

Comment (herzog31) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/page/PageImpl.java]: This property will most probably disappear with CIF-1086.

**Comment by herzog31** [comment]
@LSantha I addressed your feedback. Please check again.

**Pull request merged**